unstructured/ingest/connector/salesforce.py

```diff
@@ -0,0 +1,328 @@
+"""
+import os
+from dataclasses import dataclass
+from email.utils import formatdate
+from pathlib import Path
+from string import Template
+from textwrap import dedent
+from typing import Any, Dict, List, Type
+
+from dateutil import parser  # type: ignore
+from simple_salesforce import Salesforce
+from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+    ConnectorCleanupMixin,
+    IngestDocCleanupMixin,
+    StandardConnectorConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+class MissingCategoryError(Exception):
+    """There are no categories with that name."""
+
+
+email_template = Template(
+    """MIME-Version: 1.0
+Date: $date
+Message-ID: $message_identifier
+Subject: $subject
+From: $from_email
+To: $to_email
+Content-Type: multipart/alternative; boundary="00000000000095c9b205eff92630"
+--00000000000095c9b205eff92630
+Content-Type: text/plain; charset="UTF-8"
+$textbody
+--00000000000095c9b205eff92630
+Content-Type: text/html; charset="UTF-8"
+$textbody
+--00000000000095c9b205eff92630--
+""",
+)
+
+account_template = Template(
+    """Id: $id
+Name: $name
+Type: $account_type
+Phone: $phone
+AccountNumber: $account_number
+Website: $website
+Industry: $industry
+AnnualRevenue: $annual_revenue
+NumberOfEmployees: $number_employees
+Ownership: $ownership
+TickerSymbol: $ticker_symbol
+Description: $description
+Rating: $rating
+DandbCompanyId: $dnb_id
+""",
+)
+
+lead_template = Template(
+    """Id: $id
+Name: $name
+Title: $title
+Company: $company
+Phone: $phone
+Email: $email
+Website: $website
+Description: $description
+LeadSource: $lead_source
+Rating: $rating
+Status: $status
+Industry: $industry
+""",
+)
+
+case_template = Template(
+    """Id: $id
+Type: $type
+Status: $status
+Reason: $reason
+Origin: $origin
+Subject: $subject
+Priority: $priority
+Description: $description
+Comments: $comments
+""",
+)
+
+campaign_template = Template(
+    """Id: $id
+Name: $name
+Type: $type
+Status: $status
+StartDate: $start_date
+EndDate: $end_date
+BudgetedCost: $budgeted_cost
+ActualCost: $actual_cost
+Description: $description
+NumberOfLeads: $number_of_leads
+NumberOfConvertedLeads: $number_of_converted_leads
+""",
+)
+
+
+@dataclass
+class SimpleSalesforceConfig(BaseConnectorConfig):
+    """Connector specific attributes"""
+
+    salesforce_categories: List[str]
+    salesforce_username: str
+    salesforce_consumer_key: str
+    salesforce_private_key_path: str
+    recursive: bool = False
+
+    @staticmethod
+    def parse_folders(folder_str: str) -> List[str]:
+        """Parses a comma separated string of Outlook folders into a list."""
+        return [x.strip() for x in folder_str.split(",")]
+
+
+@dataclass
+class SalesforceIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
+    record_type: str
+    record: Dict[Any, Any]
```

Comment:
is this record truly Any, Any? want to make sure this is serializable, so hopefully the types are more constrained?

Reply:
Well key is always a string. But how would we type hint all the different value options? Bool, int, str etc.

That being said I'm going to rewrite it anyway to just be an id (str) and not a dict, and then we will GET all the records in `get_file`

We had discussed that this will create more api connections, but will be safer for especially email... :)

Reply:
Rewriting this whole part so that it is just storing Ids. Like some of the other conectors.